api/src/main/java/com/github/retrooper/packetevents/util/MojangAPIUtil.java

```diff
@@ -76,6 +77,51 @@ public static List<TextureProperty> requestPlayerTextureProperties(UUID uuid) {
         return null;
     }
 
+    /**
+     * Here you can get a full User Profile with Texture in only one HTTP Request
+     */
+    public static UserProfile createUserProfile(UUID uuid) {
```

Comment:
In general, the contents of this method should be moved to another method.
Both methods, `createUserProfile` and `requestPlayerTextureProperties` have exactly the same content, with the change that one returns a `UserProfile` and the other a list of `TextureProperty`, so just move the content to another method instead of writing the same code twice.

Reply:
The Differenc is that you can do more in a Single HTTP Request for example i need a Full User Profile,Currently i must send 2 HTTP Reqeust one to get the Name and the Other to get the Texture this is slower,more unefficent and raises the Mojang API Rate limit

Reply:
Can't you do **something** like this: https://haste.devstorage.eu/ohLf7smDAp - there was not really much difference.. https://www.diffchecker.com/Z4JFyZgY/
Just the general idea, ignore everything else